src/main/java/com/box/sdk/BoxAPIConnection.java

```diff
@@ -843,10 +843,11 @@ protected ResourceLinkType determineResourceLinkType(String resourceLink) {
         try {
             URL validUrl = new URL(resourceLink);
             String validURLStr = validUrl.toString();
-            final String apiEndpointPattern = "https://api.box.com/2.0/files/\\d+";
+            final String apiFilesEndpointPattern = "https://api.box.com/2.0/files/\\d+";
```

Comment:
```suggestion
            final String apiFilesEndpointPattern = ".*box.com/2.0/files/\\d+";
```
I didn't catch this before.  I think we should we allow for more flexibility with the subdomains, right?